_pytest/mark.py

```diff
@@ -4,17 +4,61 @@
 from operator import attrgetter
 from .compat import imap
 
+
 def alias(name):
     return property(attrgetter(name), doc='alias for ' + name)
 
 
+class ParameterSet(namedtuple('ParameterSet', 'values, marks, id')):
+    @classmethod
+    def param(cls, *value, **kw):
+        marks = kw.pop('marks', ())
+        
+        if isinstance(marks, MarkDecorator):
+            marks = marks,
+        else:
+            assert isinstance(marks, (tuple, list, set))
+
+        id = kw.pop('id', None)
+        assert not kw, kw
```

Comment:
Hmm I think it should break immediately. From my POV having `**kwargs` here is just a way to emulate the keyword-only syntax of Python 3, so it should bail if the user passes a keyword parameter we don't know about.